demo_modules/wind/wind.js

```diff
@@ -41,6 +41,26 @@ function loadJson(file) {
   });
 }
 
+function ensureNumber(num, fallback) {
+  return _.isFinite(num) || num === Infinity || num === -Infinity ? num : fallback;
+}
+
+/**
+ * @param bounds the projection bounds: [[x0, y0], [x1, y1]]
+ * @param width
+ * @param height
+ * @returns {Object} the projection bounds clamped to the specified view.
+ */
+function clampedBounds(bounds, width, height) {
+  const upperLeft = bounds[0];
+  const lowerRight = bounds[1];
+  const x = Math.max(Math.floor(ensureNumber(upperLeft[0], 0)), 0);
+  const y = Math.max(Math.floor(ensureNumber(upperLeft[1], 0)), 0);
+  const xMax = Math.min(Math.ceil(ensureNumber(lowerRight[0], width)), width - 1);
+  const yMax = Math.min(Math.ceil(ensureNumber(lowerRight[1], height)), height - 1);
+  return {x: x, y: y, xMax: xMax, yMax: yMax, width: xMax - x + 1, height: yMax - y + 1};
```

Comment:
In modern js, we can elide the value when the key and the value are the same:

{x, y, xMax, yMax, width: .., height: ...}

Reply:
Done